core/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java

```diff
@@ -121,8 +121,24 @@ public void topDocs(TopDocs topDocs, DocValueFormat[] sortValueFormats) {
         return sortValueFormats;
     }
 
-    public Aggregations aggregations() {
-        return aggregations;
+    /**
+     * Retruns <code>true</code> if this query result has unconsumed aggregations
+     */
+    public boolean hasAggs() {
+        return aggregations != null;
```

Comment:
since aggregations get nulled out when consumeAggs is called, should we compute `hasAggs = aggregations != null` in the setter? I think otherwise this might hide bugs that there are multiple places that consume aggs and all but the first one just think that there are no aggs?